### PR TITLE
PR: Add disambiguation functionality to the Python console

### LINKED ISSUE #1165: Console tabs are too big because they display the full filename
_From [andreasn...@googlemail.com](https://code.google.com/u/106330885117898830116/) on 2012-10-04T15:20:39Z_

The console tabs of running or run programs are quite big, because they display the complete filename including path, e.g.:

"Console - /home/user/python/myproject/util/module.py"

On some Spyder window setups, this big tab uses much space and thus leaves very little space for the other views tabs, I have put in the same pane. These other tabs then often show only one letter and it is somehow unpleasing to the eye.

Not so long ago, only the filename ("Console - module.py") was displayed in these tabs. Can we have an option for choosing between long and short filenames in the console tabs?

_Original issue: http://code.google.com/p/spyderlib/issues/detail?id=1165_

### PR DESCRIPTION
**Pull request by dalthviz**
Fixes #1165

### CODE REVIEW

**Comment by dalthviz** [comment]
A preview of the change:

pythonconsole

File: spyder/plugins/externalconsole.py [hunk 1]
```diff
@@ -707,22 +723,15 @@ def process_finished(self, shell_id):
...
+        """Return widget icon."""
         return ima.icon('console')
     
     def get_focus_widget(self):
-        """
-        Return the widget to give focus to when
-        this plugin's dockwidget is raised on top-level
-        """
+        """Return the widget to give focus to when
+        this plugin's dockwidget is raised on top-level"""
```
Comment (ccordoba12) [inline spyder/plugins/externalconsole.py]: Let's change it to ```"""Return the widget to give focus to."""``` to fix the style error.

**Comment by ccordoba12** [comment]
Since style tests are now passing, please squash the last four commits into one and push again :-)

**Comment by ccordoba12** [comment]
Looks good to me, thanks @dalthviz!

**Pull request merged**